diff --git a/machine/emulation.c b/machine/emulation.c
index b86fa36..5a05dc0 100644
--- a/machine/emulation.c
+++ b/machine/emulation.c
@@ -7,7 +7,7 @@
 
 static DECLARE_EMULATION_FUNC(emulate_rvc)
 {
-#ifdef __riscv_compressed
+//#ifdef __riscv_compressed
   // the only emulable RVC instructions are FP loads and stores.
 # if !defined(__riscv_flen) && defined(PK_ENABLE_FP_EMULATION)
   write_csr(mepc, mepc + 2);
@@ -61,7 +61,7 @@ static DECLARE_EMULATION_FUNC(emulate_rvc)
   } else
 #  endif
 # endif
-#endif
+//#endif
 
   return truly_illegal_insn(regs, mcause, mepc, mstatus, insn);
 }
