diff -Naur ./Makefile ../../riscvemu-2017-08-06/Makefile
--- ./Makefile	2017-08-06 09:30:12.000000000 -0300
+++ ../../riscvemu-2017-08-06/Makefile	2017-11-01 17:07:50.811434951 -0200
@@ -26,12 +26,12 @@
 # (openssl) must be installed.
 CONFIG_FS_NET=y
 # SDL support (optional)
-CONFIG_SDL=y
+# CONFIG_SDL=y
 # if set, compile the 128 bit emulator. Note: the 128 bit target does
 # not compile if gcc does not support the int128 type (32 bit hosts).
-CONFIG_INT128=y
+# CONFIG_INT128=y
 # build x86emu
-CONFIG_X86EMU=y
+# CONFIG_X86EMU=y
 # win32 build (not usable yet)
 #CONFIG_WIN32=y
 # user space network redirector
@@ -46,8 +46,9 @@
 endif
 CC=$(CROSS_PREFIX)gcc
 STRIP=$(CROSS_PREFIX)strip
-CFLAGS=-O2 -Wall -g -Werror -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -MMD
+CFLAGS=-O2 -Wall -g -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -MMD
 CFLAGS+=-D_GNU_SOURCE -DCONFIG_VERSION=\"$(shell cat VERSION)\"
+CFLAGS+=-DFLEN=0
 LDFLAGS=
 
 bindir=/usr/local/bin
diff -Naur ./riscv_cpu.c ../../riscvemu-2017-08-06/riscv_cpu.c
--- ./riscv_cpu.c	2017-08-06 09:30:12.000000000 -0300
+++ ../../riscvemu-2017-08-06/riscv_cpu.c	2017-11-01 17:06:39.331639746 -0200
@@ -47,7 +47,7 @@
 #endif
 #endif /* !FLEN */
 
-#define CONFIG_EXT_C /* compressed instructions */
+//#define CONFIG_EXT_C /* compressed instructions */
 
 //#define DUMP_INVALID_MEM_ACCESS
 //#define DUMP_MMU_EXCEPTIONS
