diff --git a/Makefile.in b/Makefile.in
index a3a2208..05b4bd5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -245,7 +245,7 @@ stamps/build-gcc-linux-stage2: $(srcdir)/riscv-gcc $(addprefix stamps/build-glib
 		@with_system_zlib@ \
 		--enable-shared \
 		--enable-tls \
-		--enable-languages=c,c++,fortran \
+		--enable-languages=c,c++ \
 		--disable-libmudflap \
 		--disable-libssp \
 		--disable-libquadmath \
@@ -289,7 +289,7 @@ stamps/build-gcc-linux-native: $(srcdir)/riscv-gcc stamps/build-gcc-linux-stage2
 		--without-system-zlib \
 		--enable-shared \
 		--enable-tls \
-		--enable-languages=c,c++,fortran \
+		--enable-languages=c,c++ \
 		--disable-libmudflap \
 		--disable-libssp \
 		--disable-libquadmath \
