diff -Naur lua-5.3.5.orig/Makefile lua-5.3.5/Makefile
--- lua-5.3.5.orig/Makefile	2016-12-20 14:26:08.000000000 -0200
+++ lua-5.3.5/Makefile	2020-05-06 13:50:04.000000000 -0300
@@ -12,11 +12,11 @@
 # LUA_ROOT, LUA_LDIR, and LUA_CDIR in luaconf.h.
 INSTALL_TOP= /usr/local
 INSTALL_BIN= $(INSTALL_TOP)/bin
-INSTALL_INC= $(INSTALL_TOP)/include
+INSTALL_INC= $(INSTALL_TOP)/include/luapp/$V
 INSTALL_LIB= $(INSTALL_TOP)/lib
 INSTALL_MAN= $(INSTALL_TOP)/man/man1
-INSTALL_LMOD= $(INSTALL_TOP)/share/lua/$V
-INSTALL_CMOD= $(INSTALL_TOP)/lib/lua/$V
+INSTALL_LMOD= $(INSTALL_TOP)/share/luapp/$V
+INSTALL_CMOD= $(INSTALL_TOP)/lib/luapp/$V
 
 # How to install. If your install program does not support "-p", then
 # you may have to run ranlib on the installed liblua.a.
@@ -38,16 +38,16 @@
 # Convenience platforms targets.
 PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris
 
+# Lua version and release.
+V= 5.3
+R= $V.5
+
 # What to install.
-TO_BIN= lua luac
+TO_BIN= luapp$V luacpp$V
 TO_INC= lua.h luaconf.h lualib.h lauxlib.h lua.hpp
-TO_LIB= liblua.a
+TO_LIB= libluapp$V.a
 TO_MAN= lua.1 luac.1
 
-# Lua version and release.
-V= 5.3
-R= $V.4
-
 # Targets start here.
 all:	$(PLAT)
 
diff -Naur lua-5.3.5.orig/src/Makefile lua-5.3.5/src/Makefile
--- lua-5.3.5.orig/src/Makefile	2018-06-25 14:46:36.000000000 -0300
+++ lua-5.3.5/src/Makefile	2020-05-06 13:50:14.000000000 -0300
@@ -6,6 +6,10 @@
 # Your platform. See PLATS for possible values.
 PLAT= none
 
+# Lua version and release.
+V= 5.3
+R= $V.5
+
 CC= gcc -std=gnu99
 CFLAGS= -O2 -Wall -Wextra -DLUA_COMPAT_5_2 $(SYSCFLAGS) $(MYCFLAGS)
 LDFLAGS= $(SYSLDFLAGS) $(MYLDFLAGS)
@@ -28,7 +32,7 @@
 
 PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris
 
-LUA_A=	liblua.a
+LUA_A=	libluapp$V.a
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
 	lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o \
 	ltm.o lundump.o lvm.o lzio.o
@@ -36,10 +40,10 @@
 	lmathlib.o loslib.o lstrlib.o ltablib.o lutf8lib.o loadlib.o linit.o
 BASE_O= $(CORE_O) $(LIB_O) $(MYOBJS)
 
-LUA_T=	lua
+LUA_T=	luapp$V
 LUA_O=	lua.o
 
-LUAC_T=	luac
+LUAC_T=	luacpp$V
 LUAC_O=	luac.o
 
 ALL_O= $(BASE_O) $(LUA_O) $(LUAC_O)
diff -Naur lua-5.3.5.orig/src/loadlib.c lua-5.3.5/src/loadlib.c
--- lua-5.3.5.orig/src/loadlib.c	2017-04-19 14:20:42.000000000 -0300
+++ lua-5.3.5/src/loadlib.c	2020-05-06 13:39:03.000000000 -0300
@@ -261,11 +261,11 @@
 ** variables that Lua check to set its paths.
 */
 #if !defined(LUA_PATH_VAR)
-#define LUA_PATH_VAR    "LUA_PATH"
+#define LUA_PATH_VAR    "LUAPP_PATH"
 #endif
 
 #if !defined(LUA_CPATH_VAR)
-#define LUA_CPATH_VAR   "LUA_CPATH"
+#define LUA_CPATH_VAR   "LUAPP_CPATH"
 #endif
 
 
diff -Naur lua-5.3.5.orig/src/lua.c lua-5.3.5/src/lua.c
--- lua-5.3.5.orig/src/lua.c	2017-04-19 14:29:57.000000000 -0300
+++ lua-5.3.5/src/lua.c	2020-05-06 13:45:51.000000000 -0300
@@ -13,6 +13,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <omp.h>
 
 #include "lua.h"
 
@@ -592,10 +593,10 @@
   return 1;
 }
 
-
 int main (int argc, char **argv) {
   int status, result;
   lua_State *L = luaL_newstate();  /* create state */
+  (void) omp_get_thread_num();
   if (L == NULL) {
     l_message(argv[0], "cannot create state: not enough memory");
     return EXIT_FAILURE;
diff -Naur lua-5.3.5.orig/src/lua.h lua-5.3.5/src/lua.h
--- lua-5.3.5.orig/src/lua.h	2018-06-13 13:58:17.000000000 -0300
+++ lua-5.3.5/src/lua.h	2020-05-06 13:39:03.000000000 -0300
@@ -21,7 +21,7 @@
 #define LUA_VERSION_NUM		503
 #define LUA_VERSION_RELEASE	"5"
 
-#define LUA_VERSION	"Lua " LUA_VERSION_MAJOR "." LUA_VERSION_MINOR
+#define LUA_VERSION	"Lua for C++ " LUA_VERSION_MAJOR "." LUA_VERSION_MINOR
 #define LUA_RELEASE	LUA_VERSION "." LUA_VERSION_RELEASE
 #define LUA_COPYRIGHT	LUA_RELEASE "  Copyright (C) 1994-2018 Lua.org, PUC-Rio"
 #define LUA_AUTHORS	"R. Ierusalimschy, L. H. de Figueiredo, W. Celes"
diff -Naur lua-5.3.5.orig/src/luaconf.h lua-5.3.5/src/luaconf.h
--- lua-5.3.5.orig/src/luaconf.h	2017-04-19 14:29:57.000000000 -0300
+++ lua-5.3.5/src/luaconf.h	2020-05-06 13:39:03.000000000 -0300
@@ -185,9 +185,9 @@
 ** In Windows, any exclamation mark ('!') in the path is replaced by the
 ** path of the directory of the executable file of the current process.
 */
-#define LUA_LDIR	"!\\lua\\"
+#define LUA_LDIR	"!\\luapp\\"
 #define LUA_CDIR	"!\\"
-#define LUA_SHRDIR	"!\\..\\share\\lua\\" LUA_VDIR "\\"
+#define LUA_SHRDIR	"!\\..\\share\\luapp\\" LUA_VDIR "\\"
 #define LUA_PATH_DEFAULT  \
 		LUA_LDIR"?.lua;"  LUA_LDIR"?\\init.lua;" \
 		LUA_CDIR"?.lua;"  LUA_CDIR"?\\init.lua;" \
@@ -195,14 +195,16 @@
 		".\\?.lua;" ".\\?\\init.lua"
 #define LUA_CPATH_DEFAULT \
 		LUA_CDIR"?.dll;" \
-		LUA_CDIR"..\\lib\\lua\\" LUA_VDIR "\\?.dll;" \
+		LUA_CDIR"..\\lib\\luapp\\" LUA_VDIR "\\?.dll;" \
 		LUA_CDIR"loadall.dll;" ".\\?.dll"
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
-#define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
-#define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
+#if !defined(LUA_ROOT)
+#define LUA_ROOT	"/opt/cartesi/"
+#endif
+#define LUA_LDIR	LUA_ROOT "share/luapp/" LUA_VDIR "/"
+#define LUA_CDIR	LUA_ROOT "lib/luapp/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
 		LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;" \
 		LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" \
